server/models/questions/addEditor.js

```diff
@@ -12,11 +12,16 @@ const { pool } = require('../database')
 function addEditor({ username, editor, questionId }) {
   return new Promise((resolve, reject) => {
     pool.query(
-      `INSERT INTO questions_editors(question_id, editor) 
-       SELECT question_id, ? FROM questions_editors WHERE editor=? AND question_id=?`,
-      [editor, username, questionId],
+      `INSERT INTO questions_editors(question_id, editor, access) 
+       SELECT question_id, ?, ? FROM questions_editors WHERE editor=? AND question_id=? AND access=?
+       ON DUPLICATE KEY UPDATE access=?`,
+      [editor, 'write', username, questionId, 'write', 'write'],
```

Comment:
1. the person **`a`** who is adding editor **`b`** should also have editor access
2. if the editor **`b`** is already there with `read` access -> then change access to `write`